### PR TITLE
(WIP) BP-28 (prototype): use Etcd as metadata store

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

This is a prototype change for demonstrating how to implement the metadata interfaces using Etcd api for BP-28 (#1113)

*BE AWARE*

This is only for prototype purpose. The initial change was done one year and half ago and re-picked to current bookkeeper repo. As @jvrao requested, pushed it out to raise the discussion.

### CODE REVIEW

**Comment by sijie** [comment]
@jvrao as we discussed today, I pushed my prototype changes here and raised a BP for discussion.

**Comment by jvrao** [comment]
I think this is waiting on the URI change. Once that is done, we can push this in. I would like some README on how to use this. Also, metadata tests to use both etcd and zk.

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by sijie** [comment]
@jvrao I assumed you approved the proposal itself, but not this change here, right? This change needs to update to complete it with URI changes.

File: metadata-stores/etcd/pom.xml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.bookkeeper</groupId>
+            <artifactId>bookkeeper-server</artifactId>
+            <version>${project.parent.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.coreos</groupId>
+            <artifactId>jetcd-core</artifactId>
+            <version>0.0.1</version>
```
Comment (eolivelli) [inline metadata-stores/etcd/pom.xml]: Nit: This version is referred in the other pom, maybe it is better to use here the variable

**Comment by sijie** [comment]
changed this to WIP, since this is not reviewable. I will update this once #1117 (BP-29) is completed.

**Pull request closed without merging**